specification/resources/resource-manager/Microsoft.Management/preview/2018-01-01-preview/management.json

```diff
@@ -0,0 +1,1266 @@
+      }
+    },
+    "EntityHierarchyItemProperties": {
+      "description": "The generic properties of a management group.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The friendly name of the management group."
+        },
+        "permissions": {
+          "title": "Permissions",
+          "$ref": "#/definitions/Permissions"
+        },
+        "children": {
+          "type": "array",
+          "description": "The list of children.",
+          "x-nullable": true,
+          "items": {
+            "$ref": "#/definitions/EntityHierarchyItem"
+          }
+        }
+      }
+    },
+    "PatchManagementGroupRequest": {
+      "description": "Management group patch parameters.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The friendly name of the management group.",
+          "x-nullable": true
+        },
+        "parentId": {
+          "type": "string",
+          "description": "(Optional) The fully qualified ID for the parent management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000",
+          "x-nullable": true
+        }
+      }
+    },
+    "CreateManagementGroupRequest": {
+      "description": "Management group creation parameters.",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the resource.  For example, /providers/Microsoft.Management/managementGroups",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the management group. For example, 00000000-0000-0000-0000-000000000000"
+        },
+        "properties": {
+          "title": "Properties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CreateManagementGroupProperties"
+        }
+      }
+    },
+    "CreateManagementGroupProperties": {
+      "description": "The generic properties of a management group used during creation.",
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The AAD Tenant ID associated with the management group. For example, 00000000-0000-0000-0000-000000000000"
+        },
+        "displayName": {
+          "type": "string",
+          "description": "The friendly name of the management group. If no value is passed then this  field will be set to the groupId.",
+          "x-nullable": true
+        },
+        "roles": {
```

Comment:
>roles [](start = 9, length = 5)

What kind of roles? It sounds as if this could be expressed as an enum

Reply:
Role definition ids

Reply:
Could this be clarified in the description?

---
In reply to: [164898347](https://github.com/Azure/azure-rest-api-specs/pull/2345#discussion_r164898347) [](ancestors = 164898347)

Reply:
Yep.